datafusion/optimizer/src/analyzer/function_rewrite.rs

```diff
@@ -0,0 +1,123 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Github renders this as a new file, but it is the old datafusion/optimizer/src/analyzer/rewrite_expr.rs with the array specific logic removed and instead invokes each `FunctionRewrite` individually